paper/paper.md

```diff
@@ -309,26 +324,20 @@ This shows how Popper helps improve the performance of scientific workflows dras
 Currently, Popper supports running MPI workloads only through Slurm using Singularity as the container engine.
-For our case study, we modified the training script to use the Horovod [@horovod] distributed deep learning framework to facilitate training with MPI [@gropp1999using] on a Slurm cluster.
+To run workflows built on the local machine inside a Slurm cluster, a config file containing Slurm specific options like the number of CPUs or nodes to use for running a job needs to be supplied to the `popper run` command.
+The user should be aware of the availability of resources like the available partitions and the free GPUs on a shared cluster to write a config file accordingly.
+The Slurm resource manager executes the image build or pull process on the Login node of a Slurm cluster and then starts executing the MPI workload on the Compute nodes using `sbatch`.
+With Singularity as the container engine, both the hybrid and bind method can be used to build a singularity container with MPI available inside it.
 The MPI based workflows can be made compatible between Slurm clusters with different MPI implementations if the Singularity images used in the workflows are built on the cluster by binding to the local MPI libraries.
 Otherwise, pre-built images with MPI installed might not be portable as the version of the MPI installed on the host and inside the image might not match.
-In addition, the programs and scripts need to be MPI compatible to utilize the total compute capacity of multiple nodes in HPC clusters.
-We recommend using a shared filesystem like NFS or AFS [@howard1988overview] mounted on each node and placing the workflow context in there to keep the workspace consistent across all the nodes.
-We used 3 VMs from Azure each with the same NVIDIA 12GB PCI P100 GPU running Ubuntu 18.04 for this experiment and used Singularity as the container engine for running this workflow.
+Additionally, the programs and scripts need to be MPI compatible to utilize the total compute capacity of multiple nodes in HPC clusters.
+In our case study, we used 3 VMs from Azure each with the same NVIDIA 12GB PCI P100 GPU running Ubuntu 18.04 for this experiment and used Singularity as the container engine for running this workflow.
 We used `mpich` which is a popular implementation of MPI, with Singularity following the bind approach, where we install MPI on the host and then bind mount the `/path/to/mpi/bin` and `/path/to/mpi/lib` of the MPI package inside the Singularity container for the MPI version in the host and the container to stay consistent.
 The training step was run using MPI on 2 compute nodes having a GPU each and the training parameters were the same as in the previous executions.
 As we can see from @Fig:casestudies, Popper allowed us to run the workflow in a Slurm cluster with MPI and hence utilize the processing power of multiple GPUs and drastically reduce the training duration.
```

Comment:
I'm not sure we can imply that from the image. This sentence can make this remark, but tweaked a bit so that it doesn't sound like the image is proof of "seamless portability"